atlasdb-autobatch/src/main/java/com/palantir/atlasdb/autobatch/Autobatchers.java

```diff
@@ -98,12 +103,33 @@ private AutobatcherBuilder(Function<Integer, EventHandler<BatchElement<I, O>>> h
             return this;
         }
 
+        public AutobatcherBuilder<I, O> safeTag(String key, String value) {
+            this.safeTags.put(key, value);
+            return this;
+        }
+
+        public AutobatcherBuilder<I, O> observability(Observability observabilityParam) {
+            this.observability = observabilityParam;
+            return this;
+        }
+
         public DisruptorAutobatcher<I, O> build() {
             Preconditions.checkArgument(purpose != null, "purpose must be provided");
             EventHandler<BatchElement<I, O>> handler = this.handlerFactory.apply(DEFAULT_BUFFER_SIZE);
 
+            EventHandler<BatchElement<I, O>> tracingHandler = (event, sequence, endOfBatch) -> {
```

Comment:
I think it would be interesting to test this to make sure that this handler does the right thing (i.e. same behaviour as before except it has the new trace).

Reply:
addressed, see new commit.